### ISSUE TITLE
Receipts attached to manually created expenses do not get SmartScanned 

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#46517]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.14-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @thienlnam 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1722292809329839

## Action Performed:
Prerequisite: Use a Control policy 

1. Submit Expense -> Manual -> Enter in an amount
2. In the confirmation page, click ‘Add Receipt’
3. Submit the expense
4. Observe the "Receipt not verified. Please confirm accuracy’." error message.

## Expected Result:
The receipt should be smartscanned in the background with the "SmartScan Everything" (SSE) feature.

## Actual Result:
The receipt does not get smartscanned in the background. 

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/a9961ce9-d99f-4b81-88e8-43bf84a4a5e0

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @rlinoz</details>

**Comment by thienlnam** [comment]
Most likely an internal issue - it seems that the receipt does not get put in the SCANREADY state and it stays in the OPEN state so it does not get scanned

**Comment by BuhusLucian** [comment]
I am not really sure if I can add this here, but I have a user that is getting this:
gmoreira@volexity.com
https://www.expensify.com/concierge/#/chat/6632349

**Comment by tgolen** [comment]
Isn't this by design? If the receipt was smartscanned, would the results be expected to overwrite the manual information that the user entered? I think that could result in just as much confusion and manual editing. @iwiznia I'm curious for your thoughts on this.

**Comment by iwiznia** [comment]
> Isn't this by design? 

If you are referring to `Receipts attached to manually created expenses do not get SmartScanned` then no, SSE (SmartScan Everything) means all receipts are scanned.

> would the results be expected to overwrite the manual information that the user entered? 

No, never overwrite the user data is one of our rules. What should happen is that you get a violation of type notice for example [this one](https://github.com/Expensify/Web-Expensify/blob/6dd23ec8a8ad3078312b77005633308cb9a4a77e/lib/Policy/Validator/Transaction.php#L791)

**Comment by iwiznia** [comment]
Why did this not have anyone assigned? 😕 
Going to reopen and add some labels

**Comment by pasyukevich** [comment]
Hi, I'm Eugene from Callstack - expert contributor group. I’d like to work on this issue.

**Comment by pasyukevich** [comment]
I wasn't able to reproduce the original issue, but for me, an error with SmartScan also occurs with report creation from receipt: 

<img width="735" alt="Image" src="https://github.com/user-attachments/assets/338ec0d9-538c-4fe8-9b2a-3373ab079065" />

**Comment by joekaufmanexpensify** [comment]
[Here](https://github.com/Expensify/App/issues/55644) is the one I found. Not sure if they are quite the same. At least the FE is different. For mine we show the receipt as scanning when we shouldn't. And then the violation after is `Receipt scan incomplete. Please verify details manually.`, which is different from the one here.

**Comment by trjExpensify** [comment]
>For mine we show the receipt as scanning when we shouldn't.

Yeah, we shouldn't be showing the receipt as scanning here either. 

>Receipt scan incomplete. Please verify details manually.

This is an old issue, the copy has been updated to yours.

**Comment by joekaufmanexpensify** [comment]
Got it, got it. Good with me to consolidate if you think they're the same RCA

**Comment by trjExpensify** [comment]
Actually, @iwiznia what is the supposed bug here? 

- I manually created an expense for £50 for Tesco 
- I attached a receipt for a totally different merchant and amount on the confirmation page 
- SSE ran in the background and added the violation message (didn't appear "Scanning..." on the front-end
- My user inputs for merchant and amount didn't change after SS was done in the background. 

This looks like it's working correctly? (Ignore the dupe, I need to be creative with my test inputs haha). 

https://github.com/user-attachments/assets/d95dc342-407e-4187-81f3-215f9e393b8a

**Comment by iwiznia** [comment]
> SSE ran in the background and added the violation message (didn't appear "Scanning..." on the front-end

This did not happen, you can see the `receipt scan incomplete message`, also I checked the receipt auditor for that receipt and it was not scanned:

![Image](https://github.com/user-attachments/assets/6a21d3b8-b4e9-49fb-8795-0317b1ca3201)

OP is correct that there is a bug here

**Comment by trjExpensify** [comment]
Interesting, when would each of these be used? 

https://github.com/Expensify/App/blob/1a8137d907323c15ff1f7125f74ce1d801cc5b7a/src/languages/en.ts#L905
https://github.com/Expensify/App/blob/1a8137d907323c15ff1f7125f74ce1d801cc5b7a/src/languages/en.ts#L5179

**Comment by trjExpensify** [comment]
I had a recollection of SSE being a Control-only feature. So I tried on a Control workspace as well, but still had the same outcome: 

<img width="1511" alt="Image" src="https://github.com/user-attachments/assets/2d021c61-e22f-4a18-8293-a00b22387016" />

**Comment by iwiznia** [comment]
> Interesting, when would each of these be used?

The copy might be a bit shitty, the names are way clearer:
receiptScanningFailed: when the smartscan failed (for example you took a pic of yourself)
receiptNotSmartScanned: receipt was not smartscanned

**Comment by trjExpensify** [comment]
Yeah, I just don't know when we intend to actually use `receiptNotSmartScanned` if SSE runs on all receipts. 🤔 

@JmillsExpensify a few questions on this: 

1. When SSE runs and the details are different than receipt, should there be a violation message under the receipt image, or just in-line the fields that have a "diff"? (e.g `amount` field for "Amount greater than SmartScan")
2. When SEE runs and the details match the receipt, what, if anything, should communicate that in UI? Is it just the "No issues found" copy above the receipt image with a green dot? 
3. Should SSE only run on Control workspaces? 
4. If SSE shouldn't run on Collect, does that mean we shouldn't have _any_ violation copy shown for an expense with a manually attached receipt? Said differently, is it wrong we're showing `receiptNotSmartScanned` in this case right now.

**Comment by iwiznia** [comment]
> Yeah, I just don't know when we intend to actually use receiptNotSmartScanned if SSE runs on all receipts. 🤔

It does not run on all receipts.

> When SSE runs and the details are different than receipt, should there be a violation message under the receipt image, or just in-line the fields that have a "diff"? (e.g amount field for "Amount greater than SmartScan")

Does not matter if it's SSE or a regular scan and IIRC the latter.

> When SEE runs and the details match the receipt, what, if anything, should communicate that in UI? Is it just the "No issues found" copy above the receipt image with a green dot?

Why would we communicate anything? IMO nothing, all is good.

> Should SSE only run on Control workspaces?

Logic is [here](https://github.com/Expensify/Web-Expensify/blob/c696e8fd874f656ab432d870a50c765ea412e7e2/lib/Policy.php#L1316) so if the property is not set, it defaults to true only for corporate workspaces.

**Comment by trjExpensify** [comment]
>It does not run on all receipts.

Which ones doesn't it run on? (Outside of the answer to the below of "it does not run on Collect").

>Does not matter if it's SSE or a regular scan and IIRC the latter.
>Why would we communicate anything? IMO nothing, all is good.

Yeah, trying to connect the dots on what's expected after looking at these in the violations doc that now look outdated: 

<img width="676" alt="Image" src="https://github.com/user-attachments/assets/a69d1367-1ee0-403c-9452-7f348ae63425" />

**Comment by iwiznia** [comment]
> Which ones doesn't it run on? (Outside of the answer to the below of "it does not run on Collect").

On policies with the property set to false or on non corporate policies without the property set

**Comment by trjExpensify** [comment]
Cool, so then yeah, my question here is still outstanding: 

>... does that mean we shouldn't have any violation copy shown for an expense with a manually attached receipt? Said differently, is it wrong we're showing receiptNotSmartScanned in this case right now.

**Comment by iwiznia** [comment]
> does that mean we shouldn't have any violation copy shown for an expense with a manually attached receipt?

No, why would it? We still want to tell approvers that the receipt was not checked by smartscan.

**Comment by trjExpensify** [comment]
Do we? That kinda' sounds like part of the DoubleCheck feature to me, really. 🤔

**Comment by trjExpensify** [comment]
Okay, so then you agree we should largely not show `receiptNotSmartScanned` on Collect policies unless they for some reason have had DoubleCheck enabled somehow?

**Comment by kadiealexander** [comment]
Is this possibly related to [this OldDot issue](https://github.com/Expensify/Expensify/issues/461968) where receipts added to manually created expenses are splitting off into their own $0 expense?

**Comment by trjExpensify** [comment]
I don't think so, sounds more like: https://expensify.slack.com/archives/CC7NECV4L/p1737998319753389

**Comment by iwiznia** [comment]
> Okay, so then you agree we should largely not show receiptNotSmartScanned on Collect policies unless they for some reason have had DoubleCheck enabled somehow?

No. The receipt was indeed not smartscanned and that's good information to show to the user, as it signifies they should review it.

**Comment by trjExpensify** [comment]
So how can receiptNotSmartScanned be [part of the double check feature](https://github.com/Expensify/App/issues/46517#issuecomment-2616361456) if you're saying it should show on workspaces where double check is not enabled?

**Comment by iwiznia** [comment]
My point is:
- Most receipts are smartscanned
- When a user creates an expense manually, we only run smartscan if the SSE setting is enabled
- When a receipt is not smartscanned, we want to let the users know, so they can review it to ensure it matches regardless of if SSE is enabled or not

**Comment by trjExpensify** [comment]
>When a receipt is not smartscanned, we want to let the users know, so they can review it to ensure it matches regardless of if SSE is enabled or not

Cool, we disagree on this point. If you want to know that, upgrade to Control and get SEE for this audit feature.

**Comment by iwiznia** [comment]
Why would we be changing that now after so many years of working like this? 
Anyway, if you want to change that, we should have a P/S to change it.

**Comment by pasyukevich** [comment]
Hi! 

Do you think I can help you with something here?

**Comment by muttmuure** [comment]
Since this is expected behavior, I'm gonna demote from CRITICAL

**Comment by trjExpensify** [comment]
It's not inconsistently reproducible or just for some users, I agree with that. 

On next steps here..

I think it's pretty clear for Control, we need to SS the receipt in the background for SSE, that doesn't seem to be happening. For Collect, I've [opened a thread](https://expensify.slack.com/archives/C03U7DCU4/p1738631317461299) in product about `receiptNotSmartScanned`.

**Comment by trjExpensify** [comment]
Alright, so the outcome here with Ioni is that we're keeping `receiptNotSmartScanned` on Collect as well. Reason being that for now, it's how you can tell automation didn't run (i.e auto-approval). 

So the outcome here is: 

- Collect is working as designed. 
- Control is not running SSE in the background when it should, so we need to fix that.

**Comment by rlinoz** [comment]
Fixing tests, will put the PR up for review today.

**Comment by rlinoz** [comment]
Deployed to staging.

**Comment by rlinoz** [comment]
Deployed to prod 🎉
